bigcode_eval/tasks/humaneval.py

```diff
@@ -57,17 +57,64 @@ def __init__(self, strip_prompt, k=[1, 10, 100], num_workers=16, timeout=3.0):
         self.num_workers = num_workers
         self.timeout = timeout
 
-    def get_dataset(self):
+        # Used for Nuggets experiments
+        self.one_shot = False
+        self.prompt_quality = None
+        self.add_context = False
+        self.args = None
+        self.doc = None
+
+    def get_dataset(self, args):
         """Returns dataset for the task or an iterable of any object, that get_prompt can handle"""
+        self.one_shot = args.one_shot
+        self.prompt_quality = args.prompt_quality
+        self.add_context = args.add_context
+        self.args = args
         return self.dataset["test"]
 
-    def get_prompt(self, doc):
-        """Builds the prompt for the LM to generate from."""
+    def get_start_context(self):
+        return "Implement answers to the following questions:\nQUESTION:\n"
+
+    def get_one_shot_example(self):
+
+        # For now, hard-coding the one shot example to be the last task in the HumanEval dataset
+        # Also hard-coding the three sample solutions (good, decent, and bad) here
+        final_sample = self.dataset["test"][-1]
+        correct_sol = final_sample["canonical_solution"]
+        really_bad_sol = "    cat: cat cat\n    dog dog dog;\n    return [giraffe if giraffe for giraffe in giraffe]"
+        decent_sol = "    lower = 2\n    upper = 8\n    return [i if i % 2 = 0 for i in range(lower, upper)]"
+        
+        # Create a list of all the different task answers possible for various experiments
+        example_task_answers = [really_bad_sol, decent_sol, correct_sol]
+
+        # If add_context is set, add additional context to the prompt. 
+        if self.add_context:
+            return final_sample["prompt"] + "\nANSWER:\n" + example_task_answers[self.prompt_quality] + "\n" + "QUESTION:\n"
```

Comment:
How did you decide on this for the context? Isn't it weird to add a random "ANSWER" in between the function signature in the prompt and the solution?

Reply:
Sorry! This was not the updated prompt. I removed the "QUESTION" and "ANSWER" portions when running experiments. The updated prompt is in the next commit.